ProcMgr Prep

nskobelevs opened the pull request:
**Arguments**
Implement argument printing for Python processes using the -a flag
Supports *varargs, keyword-only arguments and **kwargs
-a no longer prints any locals.

_Before_
<img width="300" alt="Screenshot 2021-05-24 at 10 01 12" src="https://user-images.githubusercontent.com/34107736/119326928-57342800-bc7a-11eb-9532-ee1b2b17dc65.png">

_After_
<img width="600" alt="Screenshot 2021-05-24 at 10 00 36" src="https://user-images.githubusercontent.com/34107736/119326900-500d1a00-bc7a-11eb-88d4-d95a33cf2ec0.png">

**Locals**
Locals are now printed with the -l flag

**Python 3 Source**
CMake will now auto search `/usr/src/debug/` for the Python 3 source if one is not specified.

**Misc**
Added -r to specify max recursion depth for printing locals.
Change -p to print the C stack trace if the process isn't a Python process

peadar reviewed (changes requested):
The changes look great, thanks.

One suggestion - rather than having "-p" fall back to printing the backtrace on error - maybe we can add a new argument, say "-A" or something to print "all" stacks, for both C and python. For automated tooling, that'll prove more useful. I prefer the current behaviour of having the tool fail if you ask it to print a python stack and it can't find the interpreter

nskobelevs commented:
Added the -A and reverted -p to current behaviour.
-p takes precedence so -pA is equivalent to -p

Also quick bug fix to handle null for `Reader::readString`.

Bug:
<img width="1680" alt="Screenshot 2021-05-25 at 13 57 54" src="https://user-images.githubusercontent.com/34107736/119506320-b6229b80-bd65-11eb-9503-c7e21d603719.png">

`run_mod` has a string for second argument and in this instance the argument was `NULL` so it tried reading the string at `0x0`
